### ISSUE TITLE
JSON.dump error with Fixnum hash keys

### ISSUE THREAD

**Issue by trebor8x** [ohler55/oj#399]
Converting the following hash with Fixnum-keys to JSON does not yield the desired result:
```
res = { 355 => 'first', 356 => 'second', 357 => 'third' }

JSON.dump(res)  
=> "{\"^#1\":[355,\"first\"],\"^#2\":[356,\"second\"],\"^#3\":[357,\"third\"]}"

JSON.load(JSON.dump(res))  
=> {"^#1"=>[355, "first"], "^#2"=>[356, "second"], "^#3"=>[357, "third"]}
```

Converting the hash without the `oj` gem  does not show the superfluous numbering:
```
JSON.load(JSON.dump(res))
=> {"355"=>"first", "356"=>"second", "357"=>"third"}
```

I am using oj in a rails project.
The file config/initializers/oj.rb contains:
```
Oj.optimize_rails
```

**Comment by ahorek** [comment]
Oj.optimize_rails
Oj.default_options = {:mode => :rails}

**Comment by trebor8x** [comment]
According to the [documentation](http://www.ohler.com/oj/doc/file.Rails.html) `optimize_rails` should be equal to the following:
```
require 'oj'

Oj::Rails.set_encoder()
Oj::Rails.set_decoder()
Oj::Rails.optimize()
Oj::Rails.mimic_JSON()
```

The method `mimic_JSON` is supposed to set the default mode to `compat`:
> Note that this also sets the default options of :mode to :compat and :encoding to :ascii.

However, the actual mode I observe is `custom`:
```
Oj.default_options[:mode]
=> :custom
```

Setting the mode via `Oj.default_options = {mode: :compat}` works, but I think that method `optimize_rails` should take care of it.

**Comment by ohler55** [comment]
I will take a look tonight. As for the mode, using JSON.dump does not use the mode option so it does not matter or should not matter what the mode is when calling JSON.dump.

Oj.mimic_JSON is not the same as Oj::Rails.mimic_JSON. The rails version of the json gem is not the same as the json gem without rails so a different mimic_JSON is needed.

**Comment by ohler55** [comment]
Just released Oj 3.1.3. This fixed the non-string keys in Hash encoding when using JSON.dump. It was just in dump. The generate method were fine. There was also a bug in returning the default options. The :object mode would return :custom. Both are fixed and the documentation for Oj.mimic_JSON now indicates unicode_xss instead of ascii.

Thanks for catching the Hash key bug and pointing out the doc error.

**Comment by trebor8x** [comment]
Thanks a lot for the fast fix.
